config: allow to extend the module on Tarantool CE (#8981)
tarantool/tarantool: changelogs/unreleased/config-extras.md

## feature/config

> * Allow to extend config's functionality using a non-public API on Tarantool
>   Community Edition (gh-8862).

Reviewer: ```suggestion
* Introduced a non-public API for extending the declarative config functionality
  in Tarantool Community Edition (gh-8862).
```

Author: Thanks! Updated.
